api/insight/socket.io.go

```diff
@@ -127,33 +132,49 @@ func NewSocketServer(params *chaincfg.Params, txGetter txhelpers.RawTransactionG
 
 	// Subscription to a room checks the room name is as expected for an
 	// address, joins the room, and increments the room's subscriber count.
-	server.OnEvent("/", "subscribe", func(so socketio.Conn, room string) {
+	server.OnEvent("/", "subscribe", func(so socketio.Conn, room string) string {
 		if len(room) > 64 || !isAlphaNumeric(room) {
-			return
+			return "bad address"
 		}
 		if _, err := dcrutil.DecodeAddress(room, params); err == nil {
+			// Enforce the maximum address room subscription limit.
+			numAddrSubs, _ := so.Context().(uint32)
+			if numAddrSubs >= maxAddressSubsPerConn {
+				apiLog.Warnf("Client %s failed to subscribe, at the limit.", so.ID())
+				so.Emit("error", `"too many address subscriptions"`)
+				return "too many address subscriptions"
+			}
+			numAddrSubs++
+			so.SetContext(numAddrSubs)
```

Comment:
According to go-socket.io docs, there's no need for locking the context accesses:

https://github.com/googollee/go-socket.io/blob/3f1229f016cfb51926a2ea99838fa869bd6b6246/conn.go#L25-L28